src/System.Private.CoreLib/shared/System/Globalization/ISOWeek.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using static System.DateTime;
+
+namespace System.Globalization
+{
+    public static class ISOWeek
+    {
+        private const int WeeksInLongYear = 53;
+        private const int WeeksInShortYear = 52;
+
+        private const int MinWeek = 1;
+        private const int MaxWeek = WeeksInLongYear;
+
+        public static int GetWeekOfYear(DateTime date)
+        {
+            int week = GetWeekNumber(date);
+
+            if (week < MinWeek)
```

Comment:
> if (week < MinWeek) [](start = 11, length = 20)

The following dates will return 0 from GetWeekNumber and then we'll call GetWeeksInYear(date.Year - 1); which will use year == 0 and it will throw which is wrong. 

0001/01/01
0001/01/02
0001/01/03
0001/01/04
0001/01/05
0001/01/06
0001/01/07


please add a test case in your tests for these dates

Reply:
Ah, that's a good point. Didn't have these in my tests. I'm not sure throwing is wrong, but it might throw the wrong exception or have the wrong message. What do you suggest we do in this case?

Reply:
Should GetWeekOfYear return 1 for the dates 1/1/1 to 1/1/7? I mean we shouldn't throw on these dates.

Reply:
Actually, I'm getting 1 back from `GetWeekNumber` using `0001-01-01`:

![image](https://user-images.githubusercontent.com/582487/41370751-bffe5bbc-6f48-11e8-9903-b14b93cff124.png)

Reply:
Right. My bad. I did wrong calculations in my head :-( you may ignore this. we should be fine then.